library/pool.c

```diff
@@ -254,6 +254,26 @@ static inline void make_thread_stat_id(char *buffer, int size, struct dnet_work_
 	}
 }
 
+static void dnet_update_trans_timestamp_network(struct dnet_io_req *r)
+{
+	struct dnet_net_state *st = r->st;
+	struct dnet_cmd *cmd = r->header;
+
+	if (cmd->flags & DNET_FLAGS_REPLY) {
+		struct dnet_trans *t;
+
+		/*
+		 * No need to lock @st->trans_lock, since this network thread is the only owner of the
+		 * transaction now, transaction hasn't been yet placed into IO queue.
+		 */
+		t = dnet_trans_search(st, cmd->trans);
+		if (t) {
+			dnet_trans_update_timestamp(st, t);
```

Comment:
It is not enough, it should also update 'timer' tree according to new transaction timeout.